src/main/groovy/wooga/gradle/appcenter/api/AppCenterReleaseUploader.groovy

```diff
@@ -0,0 +1,425 @@
+                }
+            } else {
+                throw new GradleException("Error while uploading chunk")
+            }
+        }
+        logger.info("upload file complete".toString())
+    }
+
+    private void uploadFinish(CreateReleaseUpload uploadCreation) {
+        logger.info("finish file upload".toString())
+        HttpPost request = new HttpPost(getUploadFinishedUri(uploadCreation))
+        HttpResponse response = client.execute(request)
+
+        if (response.statusLine.statusCode != 200) {
+            throw new GradleException("Failed to finish upload")
+        }
+
+        def responseBody = getResponseBody(response)
+        if (responseBody['error'] == true) {
+            throw new GradleException("unable to finish upload ${responseBody['message']}")
+        }
+    }
+
+    private URI getUploadReleasesUri(String uploadId = "") {
+        def builder = new URIBuilder("${API_BASE_URL}/${owner}/${applicationIdentifier}/uploads/releases/${uploadId}")
+        builder.build()
+    }
+
+    private URI getReleasesUri(String releaseId) {
+        def builder = new URIBuilder("${API_BASE_URL}/${owner}/${applicationIdentifier}/releases/${releaseId}")
+        builder.build()
+    }
+
+    private setHeaders(HttpRequest request) {
+        request.setHeader("Accept", 'application/json')
+        request.setHeader("X-API-Token", apiToken)
+    }
+
+    private onUnhandledResponse(String message, HttpResponse response) {
+        throw new GradleException("${message} [For owner(${owner}), applicationIdentifier(${applicationIdentifier})] RESPONSE STATUS CODE: ${response.statusLine.statusCode}")
+    }
+
+    private static URI getUploadSetMetadataUri(CreateReleaseUpload uploadCreation, File binary) {
+        String contentType = AppCenterRest.CONTENT_TYPES[FilenameUtils.getExtension(binary.name).toLowerCase()] ?: "application/octet-stream"
+        def builder = new URIBuilder("${uploadCreation.uploadDomain}/upload/set_metadata/${uploadCreation.packageAssetId}")
+        builder.setParameter("file_name", binary.name)
+        builder.setParameter("file_size", binary.size().toString())
+        builder.setParameter("token", uploadCreation.token)
+        builder.setParameter("content_type", contentType)
+        builder.build()
+    }
+
+    private static URI getUploadChunkUri(CreateReleaseUpload uploadCreation, Integer blockNumber) {
+        def builder = new URIBuilder("${uploadCreation.uploadDomain}/upload/upload_chunk/${uploadCreation.packageAssetId}")
+        builder.setParameter("token", uploadCreation.token)
+        builder.setParameter("block_number", blockNumber.toString())
+        builder.build()
+    }
+
+    private static URI getUploadFinishedUri(CreateReleaseUpload uploadCreation) {
+        def builder = new URIBuilder("${uploadCreation.uploadDomain}/upload/finished/${uploadCreation.packageAssetId}")
+        builder.setParameter("token", uploadCreation.token)
+        builder.build()
+    }
+
+}
```

Comment:
Check your IDE settings to add empty newline please.

Reply:
Ah. Will do. (I worked with intellij idea for this project, forgot to set it up)